src/JustSaying/AwsTools/MessageHandling/SqsMessagePublisher.cs

```diff
@@ -80,12 +85,36 @@ public async Task PublishAsync(Message message, PublishMetadata metadata, Cancel
 
     private SendMessageRequest BuildSendMessageRequest(Message message, PublishMetadata metadata)
     {
+        var messageBody = GetMessageInContext(message);
+
+        string contentEncoding = null;
+        if (CompressionOptions?.CompressionEncoding is { } compressionEncoding && CompressionRegistry is not null)
+        {
+            var bodyByteLength = Encoding.UTF8.GetByteCount(messageBody); // We should probably also include the length of the message attributes
+            if (bodyByteLength > (CompressionOptions?.MessageLengthThreshold ?? int.MaxValue)) // Well under 256KB
```

Comment:
Similar comment here - maybe there's an internal extension method we could add to simplify it and share the logic between the two sites?